developer-hub/src/tutorial/currency_simple/mod.rs

```diff
@@ -0,0 +1,733 @@
+#![doc = docify::embed!("./src/tutorial/currency_simple/mod.rs", mint_works)]
+//!
+//! It is always a good idea to build a mental model where you write *at least* one test for each
+//! "success path" of a dispatchable, and one test for each "failure path", such as:
+#![doc = docify::embed!("./src/tutorial/currency_simple/mod.rs", transfer_from_non_existent_fails)]
+//!
+//! We leave it up to you to write a test that triggers to `InsufficientBalance` error.
+//!
+//! ### Event and Error
+//!
+//! Our pallet is mainly missing two parts that are common in most FRAME pallets: Events, and
+//! Errors. First, let's understand what each are.
+//!
+//! - **Error**: The static string-based error scheme we used so far is good for readability, but it
+//!   has a few drawbacks. These string literals will bloat the final wasm blob, and are relatively
+//!   heavy to transmit and encode/decode. Moreover, it is easy to mistype then by one character.
+//!   FRAME errors are exactly a solution to maintain readability, whilst fixing the drawbacks
+//!   mentioned. In short, we use an enum to represent different variants of our error. These
+//!   variants are then mapped in an efficient way (using inly `u8` indices) to
+//!   [`sp_runtime::DispatchError::Module`] Read more about this in [`frame::pallet_macros::error`].
+//!
+//! - **Event**: Events are akin to the return type of dispatch-ables. They should represent what
+//!   happened at the end of a dispatch operation. Therefore, the convention is to use passive tense
+//!   for event names (eg. `SomethingHappened`). This allows other sub-systems or external parties
+//!   (eg. a light-client, A DApp) to listen to particular events happening, without needing to
+//!   re-execute the whole state transition function.
+//!
+//! TODO: both need to be improved a lot at the pallet-macro rust-doc level. Also my explanation
+//! of event is probably not the best.
+//!
+//! With the explanation out of the way, let's see how these components can be added. Both follow a
+//! fairly familiar syntax: normal Rust enums, with an extra `#[frame::event/error]` attribute
+//! attached.
+#![doc = docify::embed!("./src/tutorial/currency_simple/mod.rs", Event)]
+#![doc = docify::embed!("./src/tutorial/currency_simple/mod.rs", Error)]
+//!
+//! One slightly custom part of this is the `#[pallet::generate_deposit(pub(super) fn
+//! deposit_event)]` part. Without going into too much detail, in order for a pallet to emit events
+//! to the rest of the system, it needs to do two things:
+//!
+//! 1.Declare a type in its `Config` that refers to the over-arching event type of the runtime. In
+//! short, by doing this, the pallet is expressing an important bound: `type RuntimeEvent:
+//! From<Event<Self>>`. Read: There exists a `RuntimeEvent`, and it can be created from the local
+//! `enum Event` of this pallet. This enables the pallet to convert its `Event` into `RuntimeEvent`,
+//! and store it where needed.
```

Comment:
```suggestion
//! From<Event<Self>>`. Read: a `RuntimeEvent` exists, and it can be created from the local `enum
//! Event` of this pallet. This enables the pallet to convert its `Event` into `RuntimeEvent`, and
//! store it where needed.
```